compilers/concrete-compiler/compiler/lib/Common/Transformers.cpp

```diff
@@ -184,9 +184,51 @@ Result<TransportValueVerifier> getObliviousTransportValueVerifier() {
   };
 }
 
-Result<TransportValueVerifier>
-getTransportValueVerifier(const Message<concreteprotocol::GateInfo> &gateInfo) {
+Message<concreteprotocol::GateInfo>
+updateGateInfoAccordingValue(Message<concreteprotocol::GateInfo> &gate,
+                             const TransportValue &value) {
+
+  auto gateReader = gate.asReader();
+  auto gateTypeInfo = gateReader.getTypeInfo();
+  if (!gateTypeInfo.hasLweCiphertext()) {
+    return gate;
+  }
+  auto gateCiphertext = gateTypeInfo.getLweCiphertext();
+  auto gateCompression = gateCiphertext.getCompression();
+  if (gateCompression == concreteprotocol::Compression::SEED) {
+    auto valueReader = value.asReader();
+    auto valueTypeInfo = valueReader.getTypeInfo();
+    if (!valueTypeInfo.hasLweCiphertext()) {
+      return gate;
+    }
+    auto valueCiphertext = valueTypeInfo.getLweCiphertext();
+    auto valueCompression = valueCiphertext.getCompression();
+    if (valueCompression == concreteprotocol::Compression::NONE) {
+      // If the compression of transportValue is none and the gateInfo have
+      // compression we update the gateInfo to allow uncompressed transportValue
+      auto gateBuilder = gate.asBuilder();
+      gateBuilder.getTypeInfo().getLweCiphertext().setCompression(
+          concreteprotocol::Compression::NONE);
+      auto gateDimensions = gateBuilder.getRawInfo().getShape().getDimensions();
+      auto lweSize = gateCiphertext.getEncryption().getLweDimension() + 1;
+      gateDimensions.set(gateDimensions.size() - 1, lweSize);
+      auto concreteShapeDimensions = gateBuilder.getTypeInfo()
+                                         .getLweCiphertext()
+                                         .getConcreteShape()
+                                         .getDimensions();
+      concreteShapeDimensions.set(concreteShapeDimensions.size() - 1, lweSize);
+      return gateBuilder.asReader();
+    }
+  }
+  return gate;
+}
+
+Result<TransportValueVerifier> getTransportValueVerifier(
+    Message<concreteprotocol::GateInfo> &originalGateInfo) {
   return [=](const TransportValue &transportVal) -> Result<void> {
+    auto copyGateInfo = originalGateInfo;
+    auto gateInfo = updateGateInfoAccordingValue(copyGateInfo, transportVal);
```

Comment:
If we end up having more cases like that, I think it would probably be better to infer a set of valid transportvalue informations from the gateinfo (instead of statically setting it as we do now), and check against those. For the time being It is good like this !